packages/opencensus-instrumentation-grpc/src/grpc.ts

```diff
@@ -0,0 +1,478 @@
+        rootSpan.addAttribute(
+            GrpcPlugin.ATTRIBUTE_GRPC_STATUS_CODE, err.code.toString());
+        rootSpan.addAttribute(GrpcPlugin.ATTRIBUTE_GRPC_ERROR_NAME, err.name);
+        rootSpan.addAttribute(
+            GrpcPlugin.ATTRIBUTE_GRPC_ERROR_MESSAGE, err.message);
+      } else {
+        rootSpan.status = plugin.traceStatus(grpcModule.status.OK);
+        rootSpan.addAttribute(
+            GrpcPlugin.ATTRIBUTE_GRPC_STATUS_CODE,
+            grpcModule.status.OK.toString());
+      }
+      rootSpan.end();
+      return callback(err, result, trailer, flags);
+    }
+
+    plugin.tracer.wrapEmitter(call);
+    return original.call(self, call, patchedCallback);
+  }
+
+  /**
+   * Handler Server Stream and Bidirectional Stream Calls
+   */
+  private serverStreamAndBidiHandler(
+      plugin: GrpcPlugin, rootSpan: types.RootSpan, call: ServerCallWithMeta,
+      // tslint:disable-next-line:no-any
+      original: grpcModule.handleCall<any, any>,
+      // tslint:disable-next-line:no-any
+      self: grpcModule.handleCall<any, any>) {
+    let spanEnded = false;
+    const endSpan = () => {
+      if (!spanEnded) {
+        spanEnded = true;
+        rootSpan.end();
+      }
+    };
+
+    plugin.tracer.wrapEmitter(call);
+    call.on('finish', () => {
+      rootSpan.status = plugin.traceStatus(call.status.code);
+      rootSpan.addAttribute(
+          GrpcPlugin.ATTRIBUTE_GRPC_STATUS_CODE, call.status.code.toString());
+      // if there is an error, span is ended on error event, otherwise here
+      if (call.status.code === 0) {
+        endSpan();
+      }
+    });
+
+    call.on('error', (err: grpcModule.ServiceError) => {
+      rootSpan.addAttribute(GrpcPlugin.ATTRIBUTE_GRPC_ERROR_NAME, err.name);
+      rootSpan.addAttribute(
+          GrpcPlugin.ATTRIBUTE_GRPC_ERROR_MESSAGE, err.message);
+      endSpan();
+    });
+
+    return original.call(self, call);
+  }
+
+  /**
+   * Returns a function that patches the gRPC makeClientConstructor in order
+   * to patch all client methods.
+   * @returns {Function}  The register patched function.
+   */
+  private patchClient() {
+    const plugin = this;
+    return (original: MakeClientConstructor) => {
+      plugin.logger.debug('patchClient');
+      return function makeClientConstructor(
+          // tslint:disable-next-line:no-any
+          methods: grpcModule.ServiceDefinition<any>, serviceName: string,
+          options: grpcModule.GenericClientOptions) {
+        const client = original.apply(this, arguments);
+        shimmer.massWrap(
+            client.prototype, Object.keys(methods) as never[],
+            plugin.patchAllClientsMethods());
+        return client;
+      };
+    };
+  }
+
+
+  /**
+   * This function starts a span (child or root) immediately before the
+   * client method is invoked, and ends it either in a callback or stream
+   * event handler, depending on the method type.
+   */
+  private patchAllClientsMethods() {
```

Comment:
This is a misnomer -- `patchClientMethod` would be more accurate. (Or, `createClientPatchMethod`, following convention that I've stated preference for in other PRs.)

Reply:
patchServer -> getPatchServer
patchClient -> getPatchClient
patchAllClientsMethods -> getPatchedClientMethods